grovedb/src/batch/multi_insert_cost_tests.rs

```diff
@@ -0,0 +1,270 @@
+                    vec![b"key2".to_vec()],
+                )),
+                None,
+                Some(&tx),
+            )
+            .cost;
+        let non_batch_cost = non_batch_cost_1
+            .add(non_batch_cost_2)
+            .add(non_batch_cost_3)
+            .add(non_batch_cost_4);
+        tx.rollback().expect("expected to rollback");
+        let ops = vec![
+            GroveDbOp::insert_run_op(vec![], b"key1".to_vec(), Element::empty_tree()),
+            GroveDbOp::insert_run_op(
+                vec![b"key1".to_vec()],
+                b"key2".to_vec(),
+                Element::new_item_with_flags(b"pizza".to_vec(), Some([0, 1].to_vec())),
+            ),
+            GroveDbOp::insert_run_op(
+                vec![b"key1".to_vec()],
+                b"key3".to_vec(),
+                Element::empty_tree(),
+            ),
+            GroveDbOp::insert_run_op(
+                vec![b"key1".to_vec(), b"key3".to_vec()],
+                b"key4".to_vec(),
+                Element::new_reference(UpstreamFromElementHeightReference(
+                    1,
+                    vec![b"key2".to_vec()],
+                )),
+            ),
+        ];
+        let cost = db
+            .apply_batch(ops, None, Some(&tx))
+            .cost_as_result()
+            .expect("expected to apply batch");
+        assert_eq!(
+            non_batch_cost.storage_cost.added_bytes,
+            cost.storage_cost.added_bytes
+        );
+        assert_eq!(non_batch_cost.storage_cost.removed_bytes, NoStorageRemoval);
+        assert_eq!(cost.storage_cost.removed_bytes, NoStorageRemoval);
+    }
+
+    #[test]
+    fn test_batch_root_two_insert_tree_cost_same_level() {
+        let db = make_empty_grovedb();
+        let tx = db.start_transaction();
+
+        let ops = vec![
+            GroveDbOp::insert_run_op(vec![], b"key1".to_vec(), Element::empty_tree()),
+            GroveDbOp::insert_run_op(vec![], b"key2".to_vec(), Element::empty_tree()),
+        ];
+        let cost_result = db.apply_batch(ops, None, Some(&tx));
+        cost_result.value.expect("expected to execute batch");
+        let cost = cost_result.cost;
+        // Explanation for 214 storage_written_bytes
+
+        // Key -> 37 bytes
+        // 32 bytes for the key prefix
+        // 4 bytes for the key
+        // 1 byte for key_size (required space for 36)
+
+        // Value -> 37
+        //   1 for the flag option (but no flags)
+        //   1 for the enum type
+        //   1 for empty tree value
+        // 32 for node hash
+        // 0 for value hash
+        // 2 byte for the value_size (required space for 98 + up to 256 for child key)
+
+        // Parent Hook -> 39
+        // Key Bytes 4
+        // Hash Size 32
+        // Key Length 1
+        // Child Heights 2
+
+        // Total (37 + 37 + 39) * 2 = 226
+        assert_eq!(
+            cost,
+            OperationCost {
+                seek_count: 2, // todo: this seems too little
```

Comment:
Do we need to verify this first?

Reply:
I don't think so, all the verification of processing costs we can do later on. It is possible that it is just 2 because the tree was just created so it would be known to be empty. But I'm not sure because I would have thought there would be a check first to make sure it was empty which would be one more seek towards root storage.